File: BE/MediSchedule.Domain/Entities/Appointment.cs
using System.ComponentModel.DataAnnotations.Schema;
using MediSchedule.Domain.Common;
using MediSchedule.Domain.ValueObjects;

namespace MediSchedule.Domain.Entities;

public class Appointment : BaseEntity
{
    // public Guid PatientId { get; set; }
    // [ForeignKey("PatientId")]
    // public Patient Patient { get; set; } = null!;
    public Guid DoctorId { get; set; }
    [ForeignKey("DoctorId")]
    public Doctor Doctor { get; set; } = null!;
    public Guid SlotId { get; set; }
    [ForeignKey("SlotId")]
    public Slot Slot { get; set; } = null!;

    public string FullName { get; set; } = null!;
    public string Phone { get; set; } = null!;
    public string Email { get; set; } = null!;
    public DateTime AppointmentDate { get; set; }
    public TimeSpan AppointmentTime { get; set; }
    public string? FileUrl { get; set; }
    
    public string Reason { get; set; } = null!;
    public AppointmentStatus Status { get; set; } = AppointmentStatus.Pending;
}

File: fe/src/types/appointment.ts
export interface CreateAppointmentRequest {
    doctorId: string
    slotId: string
    patientName: string
    patientPhone: string
    patientEmail: string
    appointmentDate: string
    appointmentTime: string
    reason: string
    symptoms?: string
    orderId: string
}

export interface CreateAppointmentResponse {
    id: string;
}

export interface AppointmentData {
    appointmentId: string;
    appointmentCode: string
    doctor: string;
    slotId: string;
    doctorId: string;
    fullName: string;
    phone: string;
    email: string;
    date: string;
    time: string;
    symptoms: string;
    specialty: string;
    specialtyPrice: number;
    serviceFee: number;
    totalAmount: number;
    orderId: string | undefined;
}

export interface BookingData {
    appointmentId: string;
    appointmentCode: string
    patientName: string | undefined
    patientPhone: string | undefined
    patientEmail: string | undefined
    doctorName: string | undefined
    specialty: string
    appointmentDate: string | undefined
    appointmentTime: string | undefined
    symptoms?: string
    amount: number
    orderId: string | undefined
    transactionId?: string
}

export interface Appointment {
    id: string
    doctorId: string
    slotId: string
    fullName: string
    phone: string
    email: string
    appointmentDate: string // ISO date string
    appointmentTime: string // (HH:mm:ss)
    fileUrl?: string
    reason: string
    status: "pending" | "confirmed" | "completed"
    doctor?: {
        id: string
        fullName: string
        specialty: string
    }
    slot?: {
        id: string
        startTime: string
        endTime: string
        isBooked: boolean
    }
}

export type AppointmentStatus = "pending" | "confirmed" | "completed" | "all"